source/module_base/module_mixing/broyden_mixing.h

```diff
@@ -0,0 +1,280 @@
+#ifndef BROYDEN_MIXING_H_
+#define BROYDEN_MIXING_H_
+#include "mixing.h"
+#include "module_base/lapack_connector.h"
+#include "module_base/matrix.h"
+#include "module_base/memory.h"
+#include "module_base/timer.h"
+#include "module_base/tool_title.h"
+
+namespace Base_Mixing
+{
+/**
+ * @brief Simplified modified broyden_mixing method.
+ *        Ref: D.D. Johnson PRB 38, 12807 (1988)
+ *        Here the weight w0 of the error of the inverse Jacobian is set to 0 and the weight wn of
+ *        the error of each previous iteration is set to same.
+ * @note  Formula:
+ *        F = n_out - n_in
+ *        dF{i} = F_{i-1} - F{i}  //different from Ref
+ *        dn_in{i} = n_in_{i-1} - n_in{i}  //different from Ref
+ *        alpha{ij} = <dF{i}, dF{j}>
+ *        beta{ij} = inv(alpha){ij}
+ *        c{mk} = <dF{k}, F{m}>
+ *        gamma{mn} = \sum_k c{mk} * beta{kn}
+ *        n{m+1} = n_in{m} + mixing_beta*F{m} - \sum_n gamma{mn} * (dn_in{n} + mixing_beta*dF{n})
+ *        mixing_data{i} = n_in{i} + mixing_beta*F{i}
+ *        n{m+1} = \sum_i coef{i} * mixing_data{i}
+ */
+class Broyden_Mixing : public Mixing
+{
+  public:
+    Broyden_Mixing(const int& mixing_ndim, const double& mixing_beta)
+    {
+        this->mixing_ndim = mixing_ndim;
+        this->data_ndim = mixing_ndim + 1;
+        this->mixing_beta = mixing_beta;
+        this->coef = std::vector<double>(mixing_ndim + 1);
+        this->beta = ModuleBase::matrix(mixing_ndim, mixing_ndim, true);
+    }
+    virtual ~Broyden_Mixing() override
+    {
+        if (F != nullptr)
+            free(F);
+        if (dF != nullptr)
+            free(dF);
+    };
+    virtual void reset() override
+    {
+        this->ndim_cal_dF = 0;
+        this->start_dF = -1;
+        this->address = nullptr;
+    }
```

Comment:
```suggestion
    }
/*
    After full implementation of Tensor container, this part will be refactored. Presently overload strategy is used.
*/
```
@denghuilu see if any related instruction can add here?